spring-rabbit/src/main/java/org/springframework/amqp/rabbit/support/micrometer/RabbitListenerObservation.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.amqp.rabbit.support.micrometer;
+
+import io.micrometer.common.docs.KeyName;
+import io.micrometer.observation.docs.DocumentedObservation;
+
+/**
+ * Spring Rabbit Observation for listeners.
+ *
+ * @author Gary Russell
+ * @since 3.0
+ *
+ */
+public enum RabbitListenerObservation implements DocumentedObservation {
+
+	/**
+	 * Observation for Rabbit listeners.
+	 */
+	LISTENER_OBSERVATION {
+
+		@Override
+		public String getName() {
+			return "spring.rabbit.listener";
+		}
+
+		@Override
+		public String getContextualName() {
+			return "RabbitListener Observation";
```

Comment:
I see this in the `PropagatingSenderTracingObservationHandler`:
```
span.name(context.getContextualName() != null ? context.getContextualName() : context.getName());
```

so, the plain name (in your case `spring.rabbit.listener`) is not going to be visible on a span.
Or there is another logic in the super interface:
```
    /**
     * Get the span name.
     * @param context handler context
     * @return name for the span
     */
    default String getSpanName(T context) {
        String name = context.getName();
        if (StringUtils.isNotBlank(context.getContextualName())) {
            name = context.getContextualName();
        }
        return SpanNameUtil.toLowerHyphen(name);
    }
```

So, we need to consult with @marcingrzejszczak to be sure where to use one or another, or both.

Reply:
The standard name is the name you'll eventually see as a metric name (e.g. `metrics.rabbitmq.listener` - if we want to align against sth like OpenTelemetry metrics https://opentelemetry.io/docs/reference/specification/metrics/semantic_conventions/instrumentation/kafka/ cc @shakuzen). The contextual name will be translated to a span name. That could be done also in accordance with OpenTelemetry trace semantic conventions (https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/messaging/#span-name) and could be set to `<destination name> <operation name>` (that's why you need to have information from the context to set the name - this is why we called it contextual name)

Reply:
Well, this sounds like that `listenerId` has to be present in this contextual name instead of the tag (or in addition to).

Reply:
Why? For the standard name that would be literally `metrics.rabbitmq.listener`. For the contextual name you'd need to pass to the context the `destination name` (queue from rabbit template?) and `operation name` like (send / receive)

Reply:
I also see that there is a `Description` in those metrics. And we indeed add something like that to the `Timer`.
But I couldn't find anything similar in the `Observation` API. 
I may guess that this `contextualName` may be mapper.
But all of these are just speculations from my ignorance.

(Sorry for hijacking this PR for not related discussion 😄 )

Reply:
These are very good questions. Since @shakuzen was even thinking about removing `description` from metrics (that was before we decided to go with 1.10 instead of 2.0) this is why the Observation doesn't have a notion of a description and we don't set the description on a metric.

Reply:
I have just noticed that as well as sender and receiver contexts, there are also `RequestReplySenderContext` and `RequestReplyReceiverContext`.

I can probably support the sender side (where we know whether we are doing request/reply), but on the receiver side, the place where we instrument the call has no idea whether or not the downstream code is request only or request/reply. That will not be easy to change.

Reply:
I see. So, the name probably has to be `messaging.rabbitmq.listener`.
And the contextual name as you said: a destination we are listening from and `receive` as an operation.
An this could be obtained via `ObservationConvention.getContextualName(T context)`.
We just have to populate that destination to the context, therefore.

Reply:
Even if we do instrument at a different place I wonder if that will change anything for tools like Zipkin. What I mean is that if we have a `RequestReply` case, that implies the kind type to be `CLIENT` and `SERVER`. Zipkin never assumed that there will be a broker in between. For messaging theoretically you should always be using the `SenderContext` and `ReceiverContext` cause those use kind `PRODUCER` and `CONSUMER` that are desgined for messaging. I think we should NOT be using the `RequestReply...` versions cause we can break the latency visualization tools (I'm talking here about tracing only of course) cc @shakuzen @jonatan-ivanov

Reply:
Yes, I also wanted to say that request-reply is an artificial feature for Messaging per se.
It is always a produce on one side and consumer on another. And then we got opposite direction for a reply process.
It might look strange for the first producer side when we block its observation waiting for a reply where ,probably, its consumer span going to look as a child...

Reply:
Ok so in fact the request-reply is a fire and forget but in the other direction. So indeed we shouldn't be using the `RequestReply...` context but the standard `Sender...` and `Receiver...` options

Reply:
I am getting a bit confused - if the contextual name must be retrieved from the context, why is there a getter on the observation itself (where there is no context).

Does this commit get us closer to what you expect?

https://github.com/spring-projects/spring-amqp/pull/1500/commits/c7d1821a47feb842ed8a13817fee758d7eb840aa

Reply:
The contextual name should be applied on the Convention, you should not override any methods on the context itself. The idea behind `ObservationConvention` is that it contains all the information about naming and tagging (keyvalues). So that if users want to change these attributes they inject their own convention instead of chanhing anything in the production code.